packages/compiler/lib/projected-names.cadl

```diff
@@ -0,0 +1,95 @@
+// Set of projections consuming the @projectedName decorator
+
+#suppress "projections-are-experimental"
+projection op#target {
```

Comment:
@bterlson do you have an opinion on the name for this projection. 
Value could be 


- Wire:
  - `json`: Configure JSON representation of data
  - `xml`: Configure XML representation of data
- Language:
  - `csharp`: Configure C# generated code
  - `python`: Configure python generated code
  - `javascript`: Configure javascript generated code
  - `swift` : Configure swift generated code
  - `c` : Configure C generated code
- Type:
  - `client`: Configure output for the client
  - `server`: Configure output for the server

Related question should we use `enum` for the known values instead of just documenting them?